pvlib/iotools/pvgis.py

```diff
@@ -594,7 +625,18 @@ def read_pvgis_tmy(filename, pvgis_format=None):
         except AttributeError:  # str/path has no .read() attribute
             with open(str(filename), 'r') as fbuf:
                 src = json.load(fbuf)
-        return _parse_pvgis_tmy_json(src)
+        data, months_selected, inputs, meta = _parse_pvgis_tmy_json(src)
+        if map_variables is None:
+            warnings.warn(
+                'PVGIS variable names will be renamed to pvlib conventions by '
+                'default starting in pvlib 0.10.0. Specify map_variables=True '
+                'to enable that behavior now, or specify map_variables=False '
+                'to hide this warning.', pvlibDeprecationWarning
+            )
+            map_variables = False
```

Comment:
Is there a reason to duplicate the warnings inside each `outputformat` branch instead of doing a single check at the start of the function?

Reply:
Nope just poor coding.